core/src/modules/apiData.ts

```diff
@@ -1,38 +1,60 @@
 import { Api } from "actionhero";
 
 export namespace APIData {
+  export function ensureArray(
+    param: { [key: string]: any } | string,
+    api?: Api
+  ) {
+    return ensureObjectOrArray<"array">(param, api);
+  }
+
   export function ensureObject(
+    param: { [key: string]: any } | string,
+    api?: Api
+  ) {
+    return ensureObjectOrArray<"object">(param, api);
+  }
+
+  function ensureObjectOrArray<t extends "array" | "object">(
     param: { [key: string]: any } | string,
     api?: Api,
     recursing = false
-  ) {
+  ): t extends "array" ? any[] : Record<string, any> {
     if (!param) {
       return null;
     } else if (Array.isArray(param)) {
       try {
-        return param.map((row) => APIData.ensureObject(row, api, true));
+        return param.map((row) => ensureObjectOrArray(row, api, true));
       } catch (error) {
         throw new Error(
           `${param} cannot be converted to JSON object (${error})`
         );
       }
     } else if (typeof param === "string") {
       try {
-        return JSON.parse(param) as { [key: string]: any };
+        return JSON.parse(param) as t extends "array"
+          ? any[]
+          : Record<string, any>;
       } catch (error) {
         if (recursing) {
-          return param;
+          return param as unknown as t extends "array"
+            ? any[]
+            : Record<string, any>;
         } else {
           throw new Error(
             `${param} cannot be converted to JSON object (${error})`
           );
         }
       }
     } else {
-      return param;
+      return param as t extends "array" ? any[] : Record<string, any>;
     }
   }
 
+  export function ensureString<T extends string>(param: unknown) {
+    return param.toString() as T;
+  }
```

Comment:
This method is interesting - we have many times where we want to ensure an input is a string but is of a tighter Type, perhaps `Model['state']`, so that's why we can pass in an additional state here